fx-src/FirefoxStorage.js

```diff
@@ -197,6 +202,39 @@ export default class FirefoxAdapter extends BaseAdapter {
     });
   }
 
+  loadDump(records) {
+    let connection = this._connection;
+    let collection_name = this.collection;
+    return Task.spawn(function* () {
+      yield connection.executeTransaction(function* doImport() {
+        for (let record of records) {
+          const params = {
+            collection_name: collection_name,
+            record_id: record.id,
+            record: JSON.stringify(record)
+          };
+          yield connection.execute(statements.importData, params);
+        }
+        const lastModified = Math.max(...records.map(record => record.last_modified));
+        const params = {
+          collection_name: collection_name
+        };
+        const previousLastModified = yield connection.execute(
+          statements.getLastModified, params).then(result => {
+            return result ? result[0].getResultByName('last_modified') : -1;
+          });
+        if (lastModified > previousLastModified) {
+          const params = {
+            collection_name: collection_name,
+            last_modified: lastModified
+          };
+          yield connection.execute(statements.saveLastModified, params);
```

Comment:
So you calculate the max of the new records' `last_modified`, query the existing metadata table to find its stored `last_modified`, compare that to the default max, and then do `INSERT OR REPLACE`.

After doing this bulk insert, the collection is guaranteed to contain data. (You short-circuited when the incoming set of records is empty, didn't you?)

So why not just do

```
UPDATE metadata SET last_modified = (SELECT MAX(last_modified) FROM records WHERE collection = :collection) WHERE collection = :collection
```

— in other words, "make sure my last_modified time matches the data I'm storing". Right?